Unified Profiles - CSV upload and UI updates [DOC-932] (#6654)
segmentio/segment-docs: src/unified-profiles/unified-profiles-workspace.md

<!---confirm w/ Cheryl that the following warning admonition is still true --->

> warning "Unable to change data source selection after this step"
> After you've selected your data source and proceeded to the next step, you can't return to this page and make a different selection. You can add additional data sources after setting up your Unified Profiles workspace. For more information about adding additional data sources after completing the Unified Profiles guided setup, see the optional [Add additional sources and destinations to your workspace](#optional-add-additional-sources-and-destinations-to-your-workspace) documentation.

1. In Unified Profiles, select a data source to get started and click **Next**.
2. Review the popup that explains how the data source connects to Segment, and click **Continue**.

## Step 2: Add connections

After you've selected the source of your customer data, set up the connections between your data source(s) and Segment. 

You can set up 1 of the following options: 
- [CSV](#csv)
- [Salesforce and a data warehouse](#salesforce-and-a-data-warehouse)
- [A data warehouse](#data-warehouse)
- [A website or mobile app source](#website-or-mobile-app)

> If your data source isn't listed on this page, click **My source isn't listed**.

Author: ```suggestion
<!---If your data source isn't listed on this page, click **My source isn't listed**.--->
```

---
title: Create a Unified Profiles Workspace
hidden: true
---
Flex users without an existing Segment workspace that includes a Unify space can create a Unified Profiles workspace and a Unify space. The Unified Profiles workspace provides limited access to Segment.

For entitlements and limitations associated with a Unified Profiles workspace, see the [Entitlements and limitations](#segment-for-flex-entitlements-and-limitations) documentation. 

## Prerequisites

Before creating a Unified Profiles workspace, you must have requested access from the [CustomerAI](https://console.twilio.com/us1/develop/flex/customerai/overview){:target="_blank"} page in your Flex Console and been accepted into the Agent Copilot and Unified Profiles beta program.

## Step 1: Select your data source

<!---confirm w/ Cheryl that the following warning admonition is still true --->

> > warning "Unable to change data source selection after this step"

Reviewer: If a user selects CSV uploader - they can go back and change to a different type of source even if they've uploaded CSVs in the past.

For other source types, they can choose a different source type IF they haven't started any of the set up for the source type they selected.

> After you've selected your data source and proceeded to the next step, you can't return to this page and make a different selection. You can add additional data sources after setting up your Unified Profiles workspace. For more information about adding additional data sources after completing the Unified Profiles guided setup, see the optional [Add additional sources and destinations to your workspace](#optional-add-additional-sources-and-destinations-to-your-workspace) documentation.

1. In Segment for Flex, select a data source to get started and click **Next**.
1. In Unified Profiles, select a data source to get started and click **Next**.
2. Review the popup that explains how the data source connects to Segment, and click **Continue**.

## Step 2: Add connections

After you've selected where your customer data is stored, you must set up the connections between your data sources and Segment. 
After you've selected the source of your customer data, set up the connections between your data source(s) and Segment. 

You can set up one of the following options: 

> - [Salesforce and a data warehouse](#salesforce-and-a-data-warehouse)

Reviewer: The Salesforce -> Profiles work isn't going to be available until later this quarter.  Not sure when this is going to be released.

> > warning "Test profiles cannot be removed from Unified Profiles"

Reviewer: Customers will be able to contact support to have profiles removed.  It's just not available as an operation within the app

> Segment recommends only uploading production-ready data to Unified Profiles.

1. On the Getting started page, click **Upload CSV**. 
2. Review the information on the Upload profiles and custom traits page. 
3. Click **Download template** to download Segment's template CSV. 
4. Open the template CSV and enter values for the fields you'd like to update identifiers and custom traits for. These values are case sensitive. If you add a new column to your CSV file, Segment adds the data to your profiles as a custom trait. 
5. Return to your Unified Profiles workspace and upload your CSV file. You can upload 1 CSV file at a time. The CSV file that you upload must contain fewer than 10,000 rows and only contain the characters outlined in the [Allowed CSV file characters](/docs/unify/csv-upload/#allowed-csv-file-characters) documentation. 
6. Click **Finish** to return to the Getting started page. 
    _(Optional)_: To upload additional CSV files, repeat steps 1-6.

> 7. When you've finished uploading your profiles, click **Add identifiers and traits** to start [Step 3: Add identifiers and traits](#step-3-add-identifiers-and-traits).

Reviewer: For the CSV upload case - we will be defaulting identifiers so customers will just be reviewing / validating the identifiers that we set.

4. Open the template CSV and enter values for the fields you'd like to update identifiers and custom traits for. These values are case sensitive. If you add a new column to your CSV file, Segment adds the data to your profiles as a custom trait. 
5. Return to your Unified Profiles workspace and upload your CSV file. You can upload 1 CSV file at a time. The CSV file that you upload must contain fewer than 10,000 rows and only contain the characters outlined in the [Allowed CSV file characters](/docs/unify/csv-upload/#allowed-csv-file-characters) documentation. 
6. Click **Finish** to return to the Getting started page. 
    _(Optional)_: To upload additional CSV files, repeat steps 1-6.
7. When you've finished uploading your profiles, click **Add identifiers and traits** to start [Step 3: Add identifiers and traits](#step-3-add-identifiers-and-traits).

1. On the Getting started with Segment page, click **Connect Salesforce**. 
### Salesforce

1. On the Getting started page, select **Connect Salesforce** and click **Next**.  
2. You are redirected to the Salesforce login screen. Sign in to Salesforce with a user that has _View all Records_ permissions. 

> 3. On the Getting started with Segment page, click **Connect data warehouse**. 

Reviewer: as mentioned above - this isn't GA yet.

---
title: Create a Unified Profiles Workspace
hidden: true
redirect_from: '/unified-profiles/segment-for-flex'
---
Flex users without an existing Segment workspace that includes a Unify space can create a Unified Profiles workspace and a Unify space. The Unified Profiles workspace provides limited access to Segment.

For entitlements and limitations associated with a Unified Profiles workspace, see the [Entitlements and limitations](#segment-for-flex-entitlements-and-limitations) documentation. 

## Prerequisites

Before creating a Unified Profiles workspace, you must have requested access from the [CustomerAI](https://console.twilio.com/us1/develop/flex/customerai/overview){:target="_blank"} page in your Flex Console and been accepted into the Agent Copilot and Unified Profiles beta program.

## Step 1: Select your data source

> <!---confirm w/ Cheryl that the following warning admonition is still true --->

Author: ```suggestion
```

> > warning "Unable to change data source selection after this step"

Author: ```suggestion
> warning "You may be unable to change data source selection after taking action"
```

> > After you've selected your data source and proceeded to the next step, you can't return to this page and make a different selection. You can add additional data sources after setting up your Unified Profiles workspace. For more information about adding additional data sources after completing the Unified Profiles guided setup, see the optional [Add additional sources and destinations to your workspace](#optional-add-additional-sources-and-destinations-to-your-workspace) documentation.

Author: ```suggestion
> For users setting up Salesforce and a data warehouse, a data warehouse, or a website or mobile app source, once you've selected your data source, proceeded to the next step, and taken action, you can't return to this page and make a different selection. Users that opted to upload CSVs can return to this page and make a different selection or upload an additional CSV. For more information about adding additional data sources after completing the Unified Profiles guided setup, see the optional [Add additional sources and destinations to your workspace](#optional-add-additional-sources-and-destinations-to-your-workspace) documentation.
```

1. In Unified Profiles, select a data source to get started and click **Next**.
2. Review the popup that explains how the data source connects to Segment, and click **Continue**.

## Step 2: Add connections

After you've selected the source of your customer data, set up the connections between your data source(s) and Segment. 

You can set up 1 of the following options: 
- [CSV](#csv)
- [Salesforce](#salesforce)
- [A data warehouse](#data-warehouse)
- [A website or mobile app source](#website-or-mobile-app)

If your data source isn't listed on this page, click **My source isn't listed**.

### CSV

> > warning "Test profiles cannot be removed from Unified Profiles"

Author: ```suggestion
> warning "Test profiles cannot be removed from Unified Profiles in your Unified Profiles workspace"
```

> > Segment recommends only uploading production-ready data to Unified Profiles.

Author: ```suggestion
> Contact Flex support to remove test profiles you uploaded to your Unified Profiles workspace.
```

> warning "You can connect to either an iOS app or an Android app during this step"
> If you need to connect additional mobile app sources to your workspace, you can do so after completing the setup process. 

1. On the Getting started page, under the Connect your mobile apps section, click **Connect Source** and select your preferred operating system. 
2. Enter a name for your source and click **Create Source**. 
3. Add the Analytics dependency to your app by following the provided instructions. When you've added the dependency to your app, click **Next**. 
4. On the "Let's test out your connection" page, select either **Skip this step** or navigate to your app, view a few screens, then return to Segment and click **Test connection**. If Segment detects screen views on your site, the Page indicator with a check mark appears. When you've verified that your snippet is successfully installed, click **Done**.
5. After Segment marks the "Add connections" tile as complete, click **Add identifiers and traits** and begin [Step 3: Add identifiers and traits](#step-3-add-identifiers-and-traits).

> ## Step 3: Add identifiers and traits

Reviewer: We are soon going to start defaulting identifiers. (I think it'll be released some time next week). Let's sync on how we want to update this to cover these defaults.

1. On the Add identifiers and traits page, click **Add identifier**. 
2. Select either **Select default identifiers** or **Create identifier** and follow the provided steps to configure your identifiers. 
3. When you've finished selecting identifiers, click **Save**.
4. On the Add identifiers and traits page, review the identifiers. If you need to make changes to an identifier, select the menu icon in the row the identifier appears in and click **Edit** or **Delete**.
4. When you're satisfied with your identifiers, click **Add computed traits**.
5. Select up to two traits and click **Save**. <br> _Segment recommends selecting **Total inbounds**, or the number of inbound attempts that resulted in a customer engagement, and **Frequent inbound channel**, which identifies the most frequently used communication channel._

> 6. _(Optional)_: Set up predictive traits by selecting the **Set up predictive traits** dropdown and clicking **Complete setup** next to one or both traits. For more information about predictive traits, see Segment's [Predictions documentation](/docs/unify/Traits/predictions/){:target="_blank"}.

Reviewer: Customers can't create predictive traits until a certain number of events have been received in the workspace, so it will be greyed out for them until they reach the threshold.  I don't think we were able to get a clear answer on what this magic number was, so we left it vague in the UI
![Screenshot 2024-06-05 at 5 17 00 PM](https://github.com/segmentio/segment-docs/assets/8780970/a7b9ec6c-1da8-42b7-880c-73181cf4ae6a)

1. On the Add identifiers and traits page, click **Add identifier**. 
2. Select either **Select default identifiers** or **Create identifier** and follow the provided steps to configure your identifiers. 
3. When you've finished selecting identifiers, click **Save**.
4. On the Add identifiers and traits page, review the identifiers. If you need to make changes to an identifier, select the menu icon in the row the identifier appears in and click **Edit** or **Delete**.
4. When you're satisfied with your identifiers, click **Add computed traits**.
5. Select up to two traits and click **Save**. <br> _Segment recommends selecting **Total inbounds**, or the number of inbound attempts that resulted in a customer engagement, and **Frequent inbound channel**, which identifies the most frequently used communication channel._

> 6. _(Optional)_: Set up predictive traits by selecting the **Set up predictive traits** dropdown and clicking **Complete setup** next to one or both traits. For more information about predictive traits, see Segment's [Predictions documentation](/docs/unify/Traits/predictions/){:target="_blank"}.

Author: ```suggestion
6. _(Optional)_: After events from your data sources populate into your downstream destinations, you can return to the guided setup to configure predictive traits. Return to the guided setup, select the **Set up predictive traits** dropdown and click **Complete setup** next to one or both traits. For more information about predictive traits, see Segment's [Predictions documentation](/docs/unify/Traits/predictions/){:target="_blank"}.
```
